flow/src/test/java/com/sequenceiq/flow/service/flowlog/FlowLogDBServiceTest.java

```diff
@@ -140,8 +141,12 @@ public void updateLastFlowLogPayload() {
 
         String payloadJson = JsonWriter.objectToJson(payload, Map.of());
         String variablesJson = JsonWriter.objectToJson(variables, Map.of());
+        String payloadJackson = JsonUtil.writeValueAsStringSilent(payload);
+        String variablesJackson = JsonUtil.writeValueAsStringSilent(variables);
         assertEquals(payloadJson, savedFlowLog.getPayload());
         assertEquals(variablesJson, savedFlowLog.getVariables());
+        assertEquals(payloadJackson, savedFlowLog.getPayloadJackson());
+        assertEquals(variablesJackson, savedFlowLog.getVariablesJackson());
     }
 
```

Comment:
Could you please cover `save()`, `saveChain()` and `terminate()` (for sake of `finalize()`) as well?

Reply:
`terminate()` calls `finalize()` with no contextParams. I assume you meant `close()`.

Reply:
Testing `save` and `saveChain` has no added value as it just checks back the properties but no logic within. So I omit that.

Reply:
> Testing `save` and `saveChain` has no added value as it just checks back the properties but no logic within.

Well, they both have changes due to the new constructor signature, so e.g. the new jackson fields could be verified in the returned `FlowLog` / `FlowChainLog` instances. Just an idea though. :smile: